pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -44,12 +44,17 @@ def on_trainer_init(
         )
 
     def init_default_checkpoint_callback(self, checkpoint_callback):
-        if checkpoint_callback is True:
+        via_callbacks = any(isinstance(c, ModelCheckpoint) for c in self.trainer.callbacks)
+        if via_callbacks:
+            rank_zero_warn(
+                "Callbacks list contains a ModelCheckpoint instance."
+                " Trainer(checkpoint_callback=...) will be ignored."
+            )
```

Comment:
maybe do this working if checkpoint_callback is True or is an instance of ModelCheckpoint.

Reply:
what if user has checkpoint_callback=False? isn't it confusing to pass checkpoint callback and it still does not turn it off?

Reply:
I wonder why anyone would do that.. But good point :]